src/lib.rs

```diff
@@ -106,7 +106,28 @@ where
         })
     }
 
-    /// Connect to the vhost-user socket and run a dedicated thread handling all requests coming
+    /// Connect to the vhost-user socket and run a dedicated thread handling
+    /// all requests coming through this socket. This runs in an infinite loop
+    /// that should be terminating once the other end of the socket (the VMM)
+    /// hangs up.
+    pub fn start_client(&mut self, socket_path: &str) -> Result<()> {
+        let mut slave_handler = SlaveReqHandler::connect(socket_path, self.handler.clone())
+            .map_err(Error::CreateSlaveReqHandler)?;
+        let handle = thread::Builder::new()
+            .name(self.name.clone())
+            .spawn(move || loop {
+                slave_handler
+                    .handle_request()
+                    .map_err(Error::HandleRequest)?;
+            })
+            .map_err(Error::StartDaemon)?;
+
+        self.main_thread = Some(handle);
+
+        Ok(())
```

Comment:
This part looks the same as the one already in start().
How about moving it to a function (e.g. start_daemon) to avoid code duplication?

Reply:
Sure I can do some factorization :)